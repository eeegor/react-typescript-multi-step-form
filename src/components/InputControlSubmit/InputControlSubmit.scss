.input-control-submit {
	&__submit {
		margin: 0 auto;
	}
}
