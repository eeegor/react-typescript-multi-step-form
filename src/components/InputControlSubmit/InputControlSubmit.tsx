/**
 * Component InputControlSubmit
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { FormSummary, InputControl, Button, Label, Info } from '..';
import './InputControlSubmit.scss';

export interface InputControlSubmitProps {
	id: string;
	name: string;
	label?: string;
	value?: string;
	info?: string;
	type?: 'submit';
	status?: 'success' | 'danger';
	required?: boolean;
	className?: string;
	formValues?: object;
	onClick?: () => void;
	gotoStep?: (step: number) => void;
}

export class InputControlSubmit extends React.Component<InputControlSubmitProps> {
	render(): JSX.Element {
		const {
			id,
			className,
			required,
			onClick,
			label,
			type,
			name,
			status,
			info,
			value,
			formValues,
			gotoStep,
		} = this.props;

		const publicFormFields = Object.assign({}, formValues);
		delete publicFormFields['submit'];

		return (
			<InputControl id={id} type={'submit'} status={status} className={classnames(className)}>
				<Label className="input-control__label" htmlFor={id || name} label={label} />
				<Info className="input-control__info" label={info} />

				{publicFormFields && (
					<FormSummary
						gotoStep={step => gotoStep && gotoStep(step)}
						formFields={publicFormFields}
					/>
				)}

				<Button
					className={`input-control-submit__submit`}
					size="large"
					type="success"
					onClick={() => onClick && onClick()}
				>
					{value || 'Submit'}
				</Button>
			</InputControl>
		);
	}
}
