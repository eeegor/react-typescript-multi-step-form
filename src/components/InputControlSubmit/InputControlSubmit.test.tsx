import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { InputControlSubmit } from './InputControlSubmit';

describe('InputControlSubmit', () => {
	const callback = jest.fn();
	const testData = {
		id: 'submit',
		name: 'submit',
		label: 'Is this data correct?',
		value: 'Confirm your details',
		info: 'All data will be handled with care',
		onClick: callback,
	};

	it('renders markup correctly', () => {
		const component = <InputControlSubmit {...testData} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('click event works', () => {
		const wrapper = mount(<InputControlSubmit {...testData} />);
		wrapper.find('.button').simulate('click');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a valid `status: danger` attribute', () => {
		const status = 'danger';
		const wrapper = mount(<InputControlSubmit {...testData} status={status} />);
		expect(wrapper.find('.input-control-submit').hasClass('input-control--danger')).toEqual(
			true
		);
	});

	it('should adopt a valid `status: success` attribute', () => {
		const status = 'success';
		const wrapper = mount(<InputControlSubmit {...testData} status={status} />);
		expect(wrapper.find('.input-control-submit').hasClass('input-control--success')).toEqual(
			true
		);
	});

	it('should render default value if no value provided', () => {
		const defaultValue = 'Submit';
		const { value, ...testDataNoValue } = testData;
		const wrapper = mount(<InputControlSubmit {...testDataNoValue} value={''} />);
		expect(wrapper.find('.button').text()).toEqual(defaultValue);
	});

	it('should render status table with correct number of elements', () => {
		const wrapper = mount(
			<InputControlSubmit
				{...testData}
				formValues={{
					name: '',
					submit: '',
				}}
			/>
		);
		const table = wrapper.find('.form-summary .table');
		expect(table.length).toEqual(1);
		expect(table.find('.table__row__label').length).toEqual(1);
	});
});
