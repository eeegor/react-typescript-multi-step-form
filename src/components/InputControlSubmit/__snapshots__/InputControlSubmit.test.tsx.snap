// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControlSubmit renders markup correctly 1`] = `
<div
  className="input-control input-control-submit"
  id="input-control-submit"
>
  <label
    className="label input-control__label"
    htmlFor="submit"
  >
    Is this data correct?
  </label>
  <small
    className="info input-control__info"
  >
    All data will be handled with care
  </small>
  <div
    className="form-summary"
  >
    <div
      className="table"
    />
  </div>
  <button
    className="button button--success button--large input-control-submit__submit"
    onClick={[Function]}
    type="button"
  >
    Confirm your details
  </button>
</div>
`;
